### DISCUSSION TITLE
?? Removed condition  'if (name === "style")' in livewire 'function patchAttributes(from2, to)'

### DISCUSSION THREAD

**Discussion by schoolshark** [livewire/livewire#9065] [1. Help] [answered]
Why was the condition if (name === "style") 
in Livewire's function patchAttributes(from2, to) 
removed?

It was useful...
Stefan

// from livewire.js:
function patchAttributes(from2, to) {
    if (from2._x_transitioning)
        return;
    if (from2._x_isShown && !to._x_isShown) {
        return;
    }
    if (!from2._x_isShown && to._x_isShown) {
        return;
    }
    let domAttributes = Array.from(from2.attributes);
    let toAttributes = Array.from(to.attributes);
    for (let i = domAttributes.length - 1; i >= 0; i--) {
        let name = domAttributes[i].name;
        if (name === "style") // this has been removed during last updates :-(
            continue;  // this had been removed during last updates :-(
        if (!to.hasAttribute(name)) {
            from2.removeAttribute(name);
        }
    }
    for (let i = toAttributes.length - 1; i >= 0; i--) {
        let name = toAttributes[i].name;
        let value = toAttributes[i].value;
        if (name === "style") // this has been removed during last updates :-(
            continue; // this had been removed during last updates :-(
        if (from2.getAttribute(name) !== value) {
            from2.setAttribute(name, value);
        }
    }
}

**Comment by joshhanley** [comment, marked as answer]
@schoolshark thanks for the report, but it's hard to read exactly what is going on and you haven't described the problem you are having.

Can you please specify which version of Livewire you were on (full patch version e.g v3.0.5) and what version you have updated to? What problem you are now facing that didn't exist before? Also can you please format your code blocks correctly (see below).

Finally can you please provide a code snippet we can copy and paste into our own apps with steps to reproduce. Be sure to include any Blade variable definitions that are used and use a little code as possible to reproduce the issue. Then, we will be happy to help.

Thanks!

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Reply by schoolshark** [reply]
Dear Josh,

sorry for my ugly code-quoting...
in some components we have code like that:

```js

// Highlight the selected cells
function selectCell(cell, customColor) {
selectedCells.add(cell); // Add the selected cell to the set of selected cells
const dataDate = cell.getAttribute('data-date'); // Retrieve the data-date attribute of the selected cell

// Select all cells with the same data-date attribute
document.querySelectorAll(`.selectable-cell[data-date="${dataDate}"]`).forEach(c => {
      
      // Ideally, we would use a predefined CSS class for styling,
      // but this is not feasible because we need to support user-defined hex colors.
      // c.classList.add(customColorClass); 
  
      // Therefore, we apply the styles directly using inline styles.
      c.style.backgroundColor = customColor;
  
      // Add the cell to the set of selected cells
      selectedCells.add(c);
  });
}
```

After publishing the new livewire.js,
it no longer worked... :.-/

Highlightning vanished after rendering.

thanx,
Stefan from Berlin

**Reply by joshhanley** [reply]
@schoolshark no worries! I have searched through the recent changes and I can't find the change you're referencing in your original post. Do you know what version of Livewire you were using before you upgraded?

**Reply by joshhanley** [reply]
@schoolshark I've done some more digging on this, and the `name === "style"` was mistakenly added in v3.5.7 (as it was only meant to be in there for debugging) in PR #8870 but it caused other bugs so it has been removed again in v3.5.10 in PR #8926.
